### ISSUE TITLE
Rewrite not persisted

### ISSUE THREAD

**Issue by joelmoss** [ruby-next/ruby-next#116]
## What did you do?

I have a custom rewriter...

```ruby
require 'ruby-next/language'

module Proscenium
  module CssModule
    class Rewriter < RubyNext::Language::Rewriters::Text
      NAME = 'proscenium-css-module'

      def safe_rewrite(source)
        source.gsub(/:@([\w_]+)/) do |_|
          context.track! self

          match = ::Regexp.last_match(1)
          pp "Proscenium::CssModule::Name.new(:@#{match}, css_module(:#{match}))"
        end
      end
    end
  end
end

RubyNext::Language.rewriters << Proscenium::CssModule::Rewriter
require 'ruby-next/language/runtime'
```

## What did you expect to happen?

This should replace `:@something` with `Proscenium::CssModule::Name.new(:@something, css_module(:something))`, and it does work on one of my Rails apps, but not in another. There are obviously differences between the two apps, but they are many and numerous, so it's impossible to run a like for like comparison.

It's almost like something else is reverting the rewrite somewhere.

## What actually happened?

With a bit of `puts` logging, I can see that the rewriter is being called and is returning the `"Proscenium::CssModule::Name.new(:@#{match}, css_module(:#{match}))"` string. But when the page is rendered, the rewrite is somehow reverted.

So this...
```ruby
:@breadcrumbs
```

Gets logged as...

```ruby
Proscenium::CssModule::Name.new(:@breadcrumbs, css_module(:breadcrumbs))
```

but is rendered as 

```ruby
:@breadcrumbs
```

as if it was unchanged.

## Environment

**Ruby version**: 3.2.2

thx

**Comment by palkan** [comment]
Let's try to figure out what's going on. Answers to the following questions could help:

- Do you use runtime rewriting? (I guess, yes, since you mentioned Rails)
- Do you have Bootsnap in one of the applications or both? If so, what is the load order of Bootsnap and Ruby Next? Does disabling Bootsnap helps?

We can also try to turn debug mode on for Ruby Next and see if it processes the source file at all:

```sh
RUBY_NEXT_DEBUG=<name_of_some_file_to_process> ...
```

**Comment by joelmoss** [comment]
1. Do you use runtime rewriting? (I guess, yes, since you mentioned Rails)

Yes.

2. Do you have Bootsnap in one of the applications or both? If so, what is the load order of Bootsnap and Ruby Next? Does disabling Bootsnap helps?

Ok, so even though bootsnap is used in both apps, commenting out the `require 'bootsnap/setup'` in boot.rb fixed it.

So now, the question is why this would be required in one app, but not the other. Should ruby-next be loaded after bootsnap?

**Comment by palkan** [comment]
> Should ruby-next be loaded after bootsnap?

Yes, in order to correctly work with Bootsnap cache, we need to know that it's being used. Technically, Ruby Next uses `require-hooks` to transpile in runtime: https://github.com/ruby-next/require-hooks?tab=readme-ov-file#bootsnap-integration
